GCRCatalogs/dc2_object.py

```diff
@@ -229,14 +236,25 @@ def _subclass_init(self, **kwargs):
             self._columns = self._generate_columns(self._datasets)
 
         bands = [col[0] for col in self._columns if len(col) == 5 and col.endswith('_mag')]
-        self._quantity_modifiers = self._generate_modifiers(self.pixel_scale, bands)
+
+        # A slightly crude way of checking for version of schema to have modelfit mag
+        # A future improvement will be to explicitly store version information in the datasets
+        # and just rely on that versioning.
+        modelfit_mag_re = re.compile('.+_modelfit_mag')
+        has_modelfit_mag = [col for col in self._columns if modelfit_mag_re.match(col)]
+        if has_modelfit_mag:
+            self._schema_version = '1.2'
+        else:
+            self._schema_version = '1.1'
+
+        self._quantity_modifiers = self._generate_modifiers(self.pixel_scale, bands, version=self._schema_version)
         self._quantity_info_dict = self._generate_info_dict(META_PATH, bands)
 
     def __del__(self):
         self.close_all_file_handles()
 
     @staticmethod
-    def _generate_modifiers(pixel_scale=0.2, bands='ugrizy'):
+    def _generate_modifiers(pixel_scale=0.2, bands='ugrizy', version=''):
```

Comment:
set default to `None`
```suggestion
    def _generate_modifiers(pixel_scale=0.2, bands='ugrizy', version=None):
```